infobip-jackson-extension-api/src/main/java/com/infobip/jackson/NamedPropertyTypeProvider.java

```diff
@@ -0,0 +1,6 @@
+package com.infobip.jackson;
+
+public interface NamedPropertyTypeProvider extends TypeProvider {
+
+    String getJsonPropertyName();
```

Comment:
This kind of flexibility doesn't make sense to me, it seems equal to just implementing the JsonTypeResolver manually on the use case site.

Reply:
Maybe in simple case, when you have one set of types (like one TypeProvider). But if you have N sets of types, you add only N NamedPropertyTypeProvider - all handled by one PresentPropertyJsonTypeResolver. Implementing it on use case side would require either N custom resolvers (all very similar) or one generic - but here resolving a target type need to be added (as it is already done in `JsonTypeResolverFactory`).